frontend/packages/dev-console/src/components/import/app/ApplicationSelector.tsx

```diff
@@ -85,9 +85,12 @@ const ApplicationSelector: React.FC<ApplicationSelectorProps> = ({
           />
         </FormGroup>
       )}
-      {(noProjectsAvailable || selectedKey.value === CREATE_APPLICATION_KEY) && (
+      {(noProjectsAvailable ||
+        !selectedKey.value ||
```

Comment:
Hmmm - I wonder if this is a big vague to the reason why. It's essentially "no applications means we want the input", where as the above logic reads "no dropdown item selected means we want the input". It's a nit for sure, but I favour being explicit. Thoughts?

Reply:
Can't seem to come up with anything more readable other than creating a function to do the check.

Reply:
Made the comment without seeing your PR updated - my suggestion is here:
https://github.com/openshift/console/pull/2815#discussion_r328706592

Reply:
The other choice could be to add another selectedKey value, like `NO_APPLICATIONS_KEY`

Reply:
> The other choice could be to add another selectedKey value, like `NO_APPLICATIONS_KEY`

I thought of that too - but I feel it's equally weird since it's not a selectable option itself. I think we should avoid making keys that are and are not used in the dropdown. Might just be unnecessarily confusing.

Reply:
agreed